src/services/google-analytics/index.ts

```diff
@@ -16,46 +16,79 @@
 
 import loadGoogleAnalytics from './loadGoogleAnalytics';
 
+type TrackPageView = {
+  page_title?: string;
+  page_location?: string;
+  page_path: string;
+};
+
+type TrackEvent = {
+  event_action: string;
+  event_category: string;
+  event_label?: string;
+  event_value?: number;
+};
+
+type SetDimension = {
+  event_name: string;
+  dimension_key: string;
+  dimension_value: string | null;
+};
+
 /**
  * The interface of the GoogleAnalytics class below
  * replicates the interface of the ReactGA library that we were using before
  */
 
 class GoogleAnalytics {
+  static googleAnalyticsKey: string;
+
   static initialize(googleAnalyticsKey: string) {
     loadGoogleAnalytics(googleAnalyticsKey);
+    this.googleAnalyticsKey = googleAnalyticsKey;
   }
 
-  static ga(...args: any[]) {
-    window.ga(...args);
+  static gtag(...args: any[]) {
+    window.gtag(...args);
   }
```

Comment:
I don't know if we are using this method at all; but as we learned, gtag doesn't accept an array of arguments; it wants the actual `Arguments` object